verifiers/internal/gha/rekor.go

```diff
@@ -339,40 +375,28 @@ func SearchValidSignedAttestation(ctx context.Context, artifactHash string, prov
 // The certificate is verified up to Fulcio, the signature is validated
 // using the certificate, and the signature generation time is checked
 // to be within the certificate validity period.
-func verifySignedAttestation(signedAtt *SignedAttestation, trustedRoot *TrustedRoot) error {
+func verifySignedAttestation(signedAtt *SignedAttestation, trustedRoot *sigstoreRoot.TrustedRoot) error {
 	cert := signedAtt.SigningCert
 	attBytes, err := cjson.MarshalCanonical(signedAtt.Envelope)
 	if err != nil {
 		return err
 	}
 	signatureTimestamp := time.Unix(*signedAtt.RekorEntry.IntegratedTime, 0)
 
-	// 1. Verify certificate chain.
-	co := &cosign.CheckOpts{
-		RootCerts:         trustedRoot.FulcioRoot,
-		IntermediateCerts: trustedRoot.FulcioIntermediates,
-		Identities: []cosign.Identity{
-			{
-				Issuer:        certOidcIssuer,
-				SubjectRegExp: certSubjectRegexp,
-			},
-		},
-		CTLogPubKeys: trustedRoot.CTPubKeys,
-	}
-	verifier, err := cosign.ValidateAndUnpackCert(signedAtt.SigningCert, co)
-	if err != nil {
+	// Verify the certificate chain, and that the certificate was valid at the time of signing.
+	if err := sigstoreVerify.VerifyLeafCertificate(signatureTimestamp, *cert, trustedRoot); err != nil {
+		fmt.Fprintf(os.Stderr, "error verifying leaf certificate with sisgtore-go: %v\n", err)
```

Comment:
I missed that this is not a 1-1 swap between APIs. `ValidateAndUnpackCert` also checks the SCT on the certificate, so you'll need to call `VerifySignedCertificateTimestamp(cert, 1 /*threshold*/, trustedRoot` as well.

Reply:
This is also missing the identity check, which happened in a deeper call in `CheckCertificatePolicy` in `ValidateAndUnpackCert`.

For sigstore-go, you'll need to create an identity, `certID := NewShortCertificateIdentity(expected-oidc-issuer, expected-oidc-issuer-regex, expected-san, expected-san-regex)`, and you can use `""` for any that you don't want to check. You can then call `certID.Verify(summary)`, where `summary := certificate.SummarizeCertificate(leafCert)` [code](https://github.com/sigstore/sigstore-go/blob/main/pkg/verify/signed_entity.go#L531)

Reply:
Thanks. Added